test/e2e/specs/editor/various/patterns.spec.js

```diff
@@ -54,15 +55,18 @@ test.describe( 'Unsynced pattern', () => {
 			.fill( 'My unsynced pattern' );
 		await page.getByLabel( 'My unsynced pattern' ).click();
 
-		const updatedContent = await editor.getEditedPostContent();
-		expect( updatedContent ).toBe(
-			`<!-- wp:paragraph -->
-<p>A useful paragraph to reuse</p>
-<!-- /wp:paragraph -->
+		// Just get the block name and content to compare as the clientIDs will be different.
+		const originalBlock = await before.map( ( block ) => ( {
+			name: block.name,
+			content: block.attributes.content,
+		} ) );
 
-<!-- wp:paragraph -->
-<p>A useful paragraph to reuse</p>
-<!-- /wp:paragraph -->`
-		);
+		const newBlocks = await editor.getBlocks();
+		expect(
+			newBlocks.map( ( block ) => ( {
+				name: block.name,
+				content: block.attributes.content,
+			} ) )
+		).toEqual( [ ...originalBlock, ...originalBlock ] );
```

Comment:
Nit: not important but maybe worth trying for simplicity's sake:

```suggestion
		await expect.poll(
			editor.getBlocks
		).toMatchObject( [ ...originalBlock, ...originalBlock ] );
```

Reply:
That didn't work for me when I tried it locally, but feel free to push an update if it works for you, otherwise I will take a closer look tomorrow - thanks for your help with this, it is my first time writing a full playwright spec so appreciate the guidance.

Reply:
I tried to fix it in [8c3a829](https://github.com/WordPress/gutenberg/pull/54892/commits/8c3a829fd7d3a87a73c9d897a682d2f3de3f1e79). Let's see if it works!